src/node/txpackagetracker.h

```diff
@@ -58,6 +59,36 @@ class TxPackageTracker {
     /** Return how many entries exist in the orphange */
     size_t OrphanageSize();
 
+    /** Received an announcement from this peer for a tx we already know is an orphan; should be
+     * called for every peer that announces the tx, even if they are not a package relay peer.
+     * The orphan request tracker will decide when to request what from which peer - use
+     * GetOrphanRequests().
+     */
+    void AddOrphanTx(NodeId nodeid, const uint256& wtxid, const CTransactionRef& tx, bool is_preferred, std::chrono::microseconds reqtime);
+
+    /** Number of packages we are working on with this peer. Includes any entries in the orphan
+     * tracker, in-flight orphan parent requests (1 per orphan regardless of how many missing
```

Comment:
If I understand correctly the rate-limiting accounting introduced here, whatever the number of orphan parent missing, there will be an in-flight orphan parent requests issued (i.e if we’re missing 10 parents, we’re going to issue 10 in-flight orphan parent requests).

I think this should be documented as multi-party applications and contracting protocols transactions issuers might have to assemble their packages in function.

Reply:
> So from a time-sensitive contract perspective, things should be strictly better than today?

I think this is a different concern I’m aiming to underscore. It’s not related about “buckets”-capture by the attackers, rather than if you have a package of {A, B, C, D} with E as a CPFP and we have rate-limiting based on the way the parents are fetched (i.e individiually rather than in batch) the equivalent package of A+E will propagate better.

So if you would like A to confirm as soon as you can because it’s a 0-conf dual-funding, you wanna probably go for A+E.

Reply:
Okay so I’ve been looking into the code, and yes to my understanding there is _no_ processing penalty at `GETPKGTXNS` reception in `ProcessMessage` than our child has 1 or N parents, we’re calling `FindTxForGetdata` to compose the response `PKGTXNS`. This invalidate this specific source of concern.

> we have rate-limiting based on the way the parents are fetched (i.e individiually rather than in batch) the equivalent package of A+E will propagate better.

I’ll maintain there is still an interpedency between package rate-limits and package compositions, though telling what to do a LN node is unclear and can be deferred to ulterior documentation.